pkg/repository/repository.go

```diff
@@ -243,15 +257,52 @@ func (r FFRepository) DeleteSegment(identifier string) {
 func (r FFRepository) isFlagOutdated(featureConfig rest.FeatureConfig) bool {
 	oldFlag, err := r.getFlagAndCache(featureConfig.Feature, false)
 	if err != nil || oldFlag.Version == nil {
-		return false
+		// If we get an error here return true to force a cache update
+		return true
 	}
 
-	return *oldFlag.Version >= *featureConfig.Version
+	return *oldFlag.Version < *featureConfig.Version
 }
 
-func (r FFRepository) areFlagsOutdated(flags ...rest.FeatureConfig) bool {
+func (r FFRepository) getFlagsAndCache(envID string, cacheable bool) ([]rest.FeatureConfig, error) {
+	flagKey := formatFlagsKey(envID)
+	flag, ok := r.cache.Get(flagKey)
+	if ok {
+		return flag.([]rest.FeatureConfig), nil
+	}
+
+	if r.storage != nil {
+		flag, ok := r.storage.Get(flagKey)
+		if ok && cacheable {
+			r.cache.Set(flagKey, flag)
+		}
+		return flag.([]rest.FeatureConfig), nil
```

Comment:
should we move this return statement inside the ok block?   So if we fail to get from storage, and weve already failked to get from the cache, we error?

Reply:
Yea I think that makes sense